### PR TITLE
OpenRCT2: fix build type

### PR DESCRIPTION
**Pull request by Skirmisher**
By default it produces a debug build, which crashes on assertions insetad of discarding them.

#### Have the results of the proposed changes been tested?
- [x] I use the packages affected by the proposed changes on a regular basis and confirm this PR works for me
- [ ] I generally don't use the affected packages but briefly tested this PR

### CODE REVIEW

File: srcpkgs/OpenRCT2/template [hunk 1]
```diff
@@ -5,10 +5,11 @@ pkgname=OpenRCT2
 _objects_version=1.0.21
 _titles_version=0.1.2c
 version=0.3.3
-revision=1
+revision=2
 build_style=cmake
 configure_args="
  -DOPENRCT2_VERSION_TAG=${version}
+ -DCMAKE_BUILD_TYPE=RelWithDebInfo
```
Comment (sgn) [inline srcpkgs/OpenRCT2/template:12]: Would you please check with `CXXFLAGS="-DNDEBUG"` instead?

Comment (Skirmisher) [reply srcpkgs/OpenRCT2/template:12]: How is that preferable to setting it via CMake build type?

Comment (sgn) [reply srcpkgs/OpenRCT2/template:12]: Some people want to build their own package with `-O1` or `-O3` and/or disable `-g`, I suppose.
Anyway, I don't know which kitchen sink would be added into `RelWithDebInfo` in the future by CMake.

Comment (Skirmisher) [reply srcpkgs/OpenRCT2/template]: Ah, I researched more and I understand the problem now. Unfortunate that CMake does not deal with compiler flag precedence well.

Comment (Skirmisher) [reply srcpkgs/OpenRCT2/template]: Okay, I confirmed `-NDEBUG` works properly.

**Pull request merged**